pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -178,3 +193,55 @@ func (i *deploymentClientImpl) listDeploymentsForConfig(namespace, configName st
 func (i *deploymentClientImpl) updateDeployment(namespace string, deployment *kapi.ReplicationController) (*kapi.ReplicationController, error) {
 	return i.updateDeploymentFunc(namespace, deployment)
 }
+
+// failureInspection inspects the given deployment configuration for any failure causes
+// and updates the status of the configuration accordingly
+func (c *DeploymentConfigController) failureInspection(config *deployapi.DeploymentConfig, outErr error) error {
```

Comment:
Agreed, the code is cleaner now that the inspection happens once at start. Updated